engine/src/main/java/org/kigalisim/engine/support/PopulationChangeRecalcStrategy.java

```diff
@@ -0,0 +1,184 @@
+        new OverridingConverterStateGetter(engine.getStateGetter());
+    UnitConverter unitConverter = new UnitConverter(stateGetter);
+    Scope scopeEffective = scope != null ? scope : engine.getScope();
+    boolean subtractRechargeEffective = subtractRecharge != null ? subtractRecharge : true;
+    String application = scopeEffective.getApplication();
+    String substance = scopeEffective.getSubstance();
+
+    if (application == null || substance == null) {
+      engine.raiseNoAppOrSubstance("recalculating population change", "");
+    }
+
+    // Get prior population
+    EngineNumber priorPopulationRaw = engine.getStream("priorEquipment", scopeEffective, null);
+    EngineNumber priorPopulation = unitConverter.convert(priorPopulationRaw, "units");
+    stateGetter.setPopulation(priorPopulation);
+
+    // Get substance sales
+    EngineNumber substanceSalesRaw = engine.getStream("sales", scopeEffective, null);
+    EngineNumber substanceSales = unitConverter.convert(substanceSalesRaw, "kg");
+
+    // Get recharge volume
+    EngineNumber rechargeVolume = engine.calculateRechargeVolume();
+
+    // Get total volume available for new units
+    BigDecimal salesKg = substanceSales.getValue();
+    BigDecimal rechargeKg = subtractRechargeEffective ? rechargeVolume.getValue() : BigDecimal.ZERO;
+    BigDecimal availableForNewUnitsKg = salesKg.subtract(rechargeKg);
+
+    // Convert to unit delta
+    EngineNumber initialChargeRaw = engine.getInitialCharge("sales");
+    EngineNumber initialCharge = unitConverter.convert(initialChargeRaw, "kg / unit");
+    BigDecimal initialChargeKgUnit = initialCharge.getValue();
+    BigDecimal deltaUnitsRaw = divideWithZero(availableForNewUnitsKg, initialChargeKgUnit);
+    BigDecimal deltaUnits = deltaUnitsRaw;
+    EngineNumber newUnitsMarginal = new EngineNumber(
+        deltaUnits.compareTo(BigDecimal.ZERO) < 0 ? BigDecimal.ZERO : deltaUnits, "units");
+
+    // Find new total
+    BigDecimal priorPopulationUnits = priorPopulation.getValue();
+    BigDecimal newUnits = priorPopulationUnits.add(deltaUnits);
+    boolean newUnitsNegative = newUnits.compareTo(BigDecimal.ZERO) < 0;
+    BigDecimal newUnitsAllowed = newUnitsNegative ? BigDecimal.ZERO : newUnits;
+    EngineNumber newUnitsEffective = new EngineNumber(newUnitsAllowed, "units");
+
+    // Save
+    engine.setStream("equipment", newUnitsEffective, null, scopeEffective, false, null);
+    engine.setStream("newEquipment", newUnitsMarginal, null, scopeEffective, false, null);
+
+    // Recalc recharge emissions
+    engine.recalcRechargeEmissions(scopeEffective);
+  }
+
+  @Override
+  public void execute(Engine target, RecalcKit kit) {
+    // Move the logic from SingleThreadEngine.recalcPopulationChange
+    OverridingConverterStateGetter stateGetter =
+        new OverridingConverterStateGetter(kit.getStateGetter().orElseThrow(
+            () -> new IllegalStateException("StateGetter required for population change"
```

Comment:
I think this is frequently repeated. Can we make getStateGetter return a realized StateGetter and have the RecalcKitBuilder throw an exception if one is not given?